include/tensor.tcc

```diff
@@ -113,140 +136,154 @@ void* Tensor<T>::data_view(const MemoryLayout mem_layout)
+    }
 
+    // Calculate size of memory buffer
     size_t n_values = 1;
     std::vector<size_t>::const_iterator it = dims.cbegin();
-    std::vector<size_t>::const_iterator it_end = dims.cend();
-    while(it!=it_end) {
-        if((*it)<=0)
-            throw std::runtime_error("All dimensions must "\
-                                     "be greater than 0.");
-        n_values*=(*it);
-        it++;
+    for ( ; it != dims.cend(); it++) {
+        if (*it <= 0) {
+            throw std::runtime_error("All dimensions must be greater than 0.");
+        }
+        n_values *= (*it);
     }
 
-    if(n_values!=this->num_values())
+    // Make sure there is space for all the data
+    // ***WS*** TBD: we should be checking each dimension here as well
+    if (n_values != num_values()) {
         throw std::runtime_error("The provided dimensions do "\
                                   "not match the size of the "\
                                   "tensor data array");
+    }
 
-    switch(mem_layout) {
-        case MemoryLayout::fortran_contiguous :
-            this->_c_to_f_memcpy((T*)data, (T*)this->_data, this->_dims);
+    // Copy over the data
+    switch (mem_layout) {
+        case MemoryLayout::fortran_contiguous:
+            _c_to_f_memcpy((T*)data, (T*)_data, _dims);
             break;
-        case MemoryLayout::contiguous :
-            std::memcpy(data, this->_data, this->_n_data_bytes());
+        case MemoryLayout::contiguous:
+            std::memcpy(data, _data, _n_data_bytes());
             break;
-        case MemoryLayout::nested : {
+        case MemoryLayout::nested: {
             size_t starting_position = 0;
-            this->_fill_nested_mem_with_data(data, dims.data(),
+            _fill_nested_mem_with_data(data, dims.data(),
                                              dims.size(),
                                              starting_position,
-                                             this->_data);
+                                             _data);
             }
             break;
+        default:
+            throw std::runtime_error(
+                "Unsupported MemoryLayout value in "\
+                "Tensor<T>.fill_mem_space().");
     }
-    return;
 }
 
+// copy values from nested memory structure to contiguous memory structure
 template <class T>
 void* Tensor<T>::_copy_nested_to_contiguous(void* src_data,
                                             const size_t* dims,
                                             const size_t n_dims,
                                             void* dest_data)
 {
-    if(n_dims > 1) {
+    if (n_dims > 1) {
         T** current = (T**)src_data;
-        for(size_t i = 0; i < dims[0]; i++) {
+        for (size_t i = 0; i < dims[0]; i++) {
           dest_data =
-            this->_copy_nested_to_contiguous(*current, &dims[1],
+            _copy_nested_to_contiguous(*current, &dims[1],
```

Comment:
Align text

Reply:
fixed. (These all came up from removing "this->", in case you're curious.)